From 5612d10bf8c35a9fa16f85cbb980efb5d9b1e12d Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Mon, 16 Nov 2015 15:22:45 -0500
Subject: [PATCH 018/134] parse-events: Fix output of %llu for 64 bit values
 read on 32 bit machines

When a long value is read on 32 bit machines for 64 bit output, the parsing
needs to change "%lu" into "%llu", as the value is read natively.

Unfortunately, if "%llu" is already there, the code will add another "l" to
it and fail to parse it properly.

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 event-parse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/event-parse.c b/event-parse.c
index 0a30253e87c8..24373d4fa6e7 100644
--- a/event-parse.c
+++ b/event-parse.c
@@ -4824,7 +4824,7 @@ static void pretty_print(struct trace_seq *s, void *data, int size, struct event
 						break;
 					}
 				}
-				if (pevent->long_size == 8 && ls &&
+				if (pevent->long_size == 8 && ls == 1 &&
 				    sizeof(long) != 8) {
 					char *p;
 
-- 
2.5.5

